context/AuthContextProvider.tsx

```diff
@@ -84,9 +75,9 @@ export const AuthContextProvider = ({
 
   /**
    * Get user data from the session
-   * @returns {Promise<void>}
+   * @returns {Promise<void | IUser>} - The user data or an error.
    */
-  const getUser = useCallback(async () => {
+  const getUser = async (): Promise<void | IUser> => {
```

Comment:
We will need unit tests for this file. I do not see a test file at all.

Reply:
roger that.  I will add one.